src/Common/src/Interop/FreeBSD/Interop.Process.cs

```diff
@@ -0,0 +1,444 @@
+                    throw new ArgumentOutOfRangeException(nameof(pid));
+                }
+
+                count = (int)bytesLength / sizeof(kinfo_proc);
+            }
+            catch
+            {
+                Marshal.FreeHGlobal((IntPtr)pBuffer);
+                throw;
+            }
+
+            return kinfo;
+        }
+
+        /// <summary>
+        /// Gets the process information for a given process
+        /// </summary>
+        /// <param name="pid">The PID (process ID) of the process</param>
+        /// <returns>
+        /// Returns a valid ProcessInfo struct for valid processes that the caller
+        /// has permission to access; otherwise, returns null
+        /// </returns>
+        static public unsafe ProcessInfo GetProcessInfoById(int pid)
+        {
+            kinfo_proc* kinfo = null;
+            int count;
+            ProcessInfo info;
+
+            // Negative PIDs are invalid
+            if (pid < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(pid));
+            }
+
+            try
+            {
+                kinfo = getProcInfo(pid, true, out count);
+                if (kinfo == null || count < 1)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(pid));
+                }
+
+                Span<kinfo_proc> process = new Span<kinfo_proc>(kinfo, count);
+
+                // Get the process information for the specified pid
+                info = new ProcessInfo();
+
+                info.ProcessName = Marshal.PtrToStringAnsi((IntPtr)kinfo->ki_comm);
+                info.BasePriority = kinfo->ki_nice;
+                info.VirtualBytes = (long)kinfo->ki_size;
+                info.WorkingSet = kinfo->ki_rssize;
+                info.SessionId = kinfo ->ki_sid;
+
+                for(int i=0; i < process.Length; i++)
+                {
+                    var ti = new ThreadInfo()
+                    {
+                        _processId = pid,
+                        _threadId = (ulong)process[i].ki_tid,
+                        _basePriority = process[i].ki_nice,
+                        _startAddress = (IntPtr)process[i].ki_tdaddr
+                    };
+                    info._threadInfoList.Add(ti);
+                }
+            }
+            finally
+            {
+                Marshal.FreeHGlobal((IntPtr)kinfo);
+            }
+
+            return info;
+        }
+
+        /// <summary>
+        /// Gets the process information for a given process
+        /// </summary>
+        // 
+        /// <param name="pid">The PID (process ID) of the process</param>
+        /// <param name="tid">The TID (thread ID) of the process</param>
+        /// <returns>
+        /// Returns basic info about thread. If tis is 0, it will return
+        /// info for process e.g. main thread.
+        /// </returns>
+        public unsafe static proc_stats getThreadInfo(int pid, int tid)
```

Comment:
Our coding convention is to use pascal casing for function names. Can you please fix all the functions that you have added and that are not following it?